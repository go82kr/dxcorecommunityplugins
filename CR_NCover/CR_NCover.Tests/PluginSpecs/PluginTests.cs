using ApprovalTests;
using CheckPoint;
using NUnit.Framework;
using TypeMock.ArrangeActAssert;
using TypeMock;

namespace CR_NCover.Tests.PluginSpecs
{
	[TestFixture]
	[Isolated]
	public class PluginTests
	{
		[Test]
		public void DontStartTheExplorerWhenThePluginIsInitialized()
		{
			Sentry pluginSentry = Monitor.Interactions(typeof(PlugIn));

			Isolate.WhenCalled(() => VisualStudio.HasASolutionOpened).WillReturn(false);
			Isolate.WhenCalled(() => VisualStudio.HasAnActiveTextView).WillReturn(true);

			PlugIn plugin = new PlugIn();

			Approvals.Approve(pluginSentry.Report);
		}

		[Test]
		public void StartTheServiceWhenASolutionIsOpen()
		{
			Sentry pluginSentry = Monitor.Interactions(typeof(PlugIn));

			Isolate.WhenCalled(() => VisualStudio.HasASolutionOpened).WillReturn(false);
			Isolate.WhenCalled(() => VisualStudio.HasAnActiveTextView).WillReturn(true);
			Isolate.WhenCalled(() => VisualStudio.Solution).WillReturn("");

			PlugIn plugin = new PlugIn();

			plugin.AfterOpeningSolution();

			Approvals.Approve(pluginSentry.Report);
		}

		[Test]
		public void HideOverlayWhenTextHasChanged()
		{
			Sentry pluginSentry = Monitor.Interactions(typeof(PlugIn));

			Isolate.WhenCalled(() => VisualStudio.HasASolutionOpened).WillReturn(false);

			PlugIn plugin = new PlugIn();

			plugin.WhenTextChanges(null);

			Approvals.Approve(pluginSentry.Report);
		}

		[Test]
		public void ShowOverlayWhenResultsUpdated()
		{
			Sentry pluginSentry = Monitor.Interactions(typeof(PlugIn));

			Isolate.WhenCalled(() => VisualStudio.Solution).WillReturn("");
			Isolate.WhenCalled(() => VisualStudio.HasASolutionOpened).WillReturn(true);
			Isolate.WhenCalled(() => VisualStudio.HasAnActiveTextView).WillReturn(true);
			PlugIn plugin = new PlugIn();

			plugin.WhenResultsUpdated();

			Approvals.Approve(pluginSentry.Report);
		}

		[Test]
		public void StartTheExplorerWhenInitializingThePlugin()
		{
			Sentry pluginSentry = Monitor.Interactions(typeof(PlugIn));

			Isolate.WhenCalled(() => VisualStudio.Solution).WillReturn("");
			Isolate.WhenCalled(() => VisualStudio.HasASolutionOpened).WillReturn(true);
			Isolate.WhenCalled(() => VisualStudio.HasAnActiveTextView).WillReturn(true);
			PlugIn plugin = new PlugIn();

			Approvals.Approve(pluginSentry.Report);
		}

		[Test]
		public void UpdateOverlayWhenPaintingBackground()
		{
			Sentry pluginSentry = Monitor.Interactions(typeof(PlugIn));

			Isolate.WhenCalled(() => VisualStudio.Solution).WillReturn("");
			Isolate.WhenCalled(() => VisualStudio.HasASolutionOpened).WillReturn(true);
			Isolate.WhenCalled(() => VisualStudio.HasAnActiveTextView).WillReturn(true);
			PlugIn plugin = new PlugIn();

			plugin.WhenPaintingBackground(null);

			Approvals.Approve(pluginSentry.Report);
		}
	}
}