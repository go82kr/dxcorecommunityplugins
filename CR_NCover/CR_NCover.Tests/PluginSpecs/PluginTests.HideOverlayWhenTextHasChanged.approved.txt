(CR_NCover.CoverageOverlay) overlay..ctor()
(CR_NCover.Service) service..ctor()
(CR_NCover.CoverageOverlay) overlay.Hide()
